mico-core/src/main/java/io/github/ust/mico/core/REST/ApplicationController.java

```diff
@@ -121,18 +121,22 @@
                 linkTo(methodOn(ApplicationController.class).getApplicationsByShortName(shortName)).withSelfRel()));
     }
 
-    @PutMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}/services")
-    public ResponseEntity addServiceToApplication(@PathVariable(PATH_VARIABLE_SHORT_NAME) String _shortName,
-                                                                         @PathVariable(PATH_VARIABLE_VERSION) String _version,
-                                                                         @RequestBody MicoService _service) {
-        Optional<MicoService> serviceOptional = serviceRepository.findByShortNameAndVersion(_service.getShortName(), _service.getVersion());
-        Optional<MicoApplication> applicationOptional = applicationRepository.findByShortNameAndVersion(_shortName, _version);
+    @PostMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}/services")
+    public ResponseEntity addServiceToApplication(@PathVariable(PATH_VARIABLE_SHORT_NAME) String applicationShortName,
+                                                  @PathVariable(PATH_VARIABLE_VERSION) String applicationVersion,
+                                                  @RequestBody MicoService serviceFromBody) {
+        Optional<MicoService> serviceOptional = serviceRepository.findByShortNameAndVersion(serviceFromBody.getShortName(), serviceFromBody.getVersion());
+        Optional<MicoApplication> applicationOptional = applicationRepository.findByShortNameAndVersion(applicationShortName, applicationVersion);
         if (serviceOptional.isPresent() && applicationOptional.isPresent()) {
-            MicoService newService = serviceOptional.get();
+            MicoService service = serviceOptional.get();
             MicoApplication application = applicationOptional.get();
-            MicoApplication applicationWithService = application.toBuilder().service(newService).build();
-            applicationRepository.save(applicationWithService);
-            return ResponseEntity.noContent().build();
+            if (!application.getServices().contains(service)) {
```

Comment:
Could use null check here instead of trying to fix lombock issue in MicoApplication constructor.

Reply:
You would have to add this check in every further usage and in every location where the list is used. Isn't it better to make sure that the list was initialized properly?

Reply:
Yes but then every instance of this problem should be refactored like this for consistency.

Reply:
I suggest, to go with a combination of both:
- make sure the list is initialized properly **AND**
- add an issue for refactoring similar issues

Reply:
#282